src/view3d.cpp

```diff
@@ -104,7 +117,15 @@ void View3D::geometryInserted(const QModelIndex& /*unused*/, int firstRow, int l
 {
     const GeometryCollection::GeometryVec& geoms = m_geometries->get();
     for (int i = firstRow; i <= lastRow; ++i)
-        geoms[i]->initializeGL();
+    {
+        if (m_boundingBoxShader->isValid())
+        {
+            geoms[i]->setShaderId("boundingbox", m_boundingBoxShader->shaderProgram().programId());
+            geoms[i]->setShaderId("meshface", m_meshFaceShader->shaderProgram().programId());
+            geoms[i]->setShaderId("meshedge", m_meshEdgeShader->shaderProgram().programId());
```

Comment:
Looks like we'd be better off having a `ShaderCache` object or some such, to keep track of the set of compiled utility shaders.  That would separate some of the extra entangled logic out of `Geometry` / `View3D`, and make a central place to manage utility shaders.